lib/pip.js

```diff
@@ -50,29 +51,39 @@ function installRequirements() {
 
     this.serverless.cli.log(`Docker Image: ${this.options.dockerImage}`);
 
-    // Check docker server os type from 'docker version'
-    let volPath;
+    // Determine os platform of docker CLI from 'docker version'
     options = [
-      'version', '--format', '{{with .Server}}{{.Os}}{{end}}'
+      'version', '--format', '{{with .Client}}{{.Os}}{{end}}'
     ];
-    const ps = spawnSync(cmd, options, {'timeout': 2000, 'encoding': 'utf-8'});
+    const ps = spawnSync(cmd, options, {'timeout': 10000, 'encoding': 'utf-8'});
     if (ps.error) {
       if (ps.error.code === 'ENOENT') {
         throw new Error('docker not found! Please install it.');
       }
       throw new Error(ps.error);
-    }
-    if (ps.status !== 0) {
+    } else if (ps.status !== 0) {
       throw new Error(ps.stderr);
-    } else if (ps.stdout.trim() === 'windows') {
-      volPath = this.servicePath.replace(/\\/g, '/').replace(/^\/mnt\//, '/');
+    }
+
+    let bindPath;
+    const cliPlatform = ps.stdout.trim();
+    if (process.platform === 'win32') {
+      bindPath = this.servicePath.replace(/\\([^\s])/g, '/$1');
+      if (cliPlatform === 'windows') {
```

Comment:
Probably not, since I'm not even 100% sure what the code is trying to attempt now. Is this correct?

1. if `process.platform` indicates windows:
   1. replace all `\` with `/` (and something whitespace, but I'm not sure what)
   1. if docker platform is windows:
      1. convert `/foo/bar/baz` to `foo:/bar/baz`
1. else-if WSL:
    1. strip `/mnt` from start of string.
    1. if docker platform is windows:
        1. convert `/foo/bar/baz` to `foo:/bar/baz`


I think the following might cleaner/clearer:
```javascript
    let bindPath = this.servicePath;
    const cliPlatform = ps.stdout.trim();
    if (process.platform === 'win32') {
      bindPath = this.servicePath.replace(/\\([^\s])/g, '/$1');
    } else if (isWsl) {
      bindPath = this.servicePath.replace(/^\/mnt\//, '/');
    }
    if (cliPlatform === 'windows') {
      bindPath = bindPath.replace(/^\/(\w)\//i, '$1:/');
    }
```

Reply:
#144